engine/src/main/java/org/terasology/rendering/dag/nodes/ToneMappingNode.java

```diff
@@ -15,41 +15,25 @@
  */
 package org.terasology.rendering.dag.nodes;
 
-import org.terasology.config.Config;
 import org.terasology.monitoring.PerformanceMonitor;
-import org.terasology.registry.In;
-import org.terasology.rendering.assets.material.Material;
 import org.terasology.rendering.dag.AbstractNode;
-import org.terasology.rendering.opengl.FBO;
-import org.terasology.rendering.opengl.FrameBuffersManager;
-import org.terasology.rendering.world.WorldRenderer;
-import static org.lwjgl.opengl.GL11.GL_COLOR_BUFFER_BIT;
-import static org.lwjgl.opengl.GL11.GL_DEPTH_BUFFER_BIT;
-import static org.lwjgl.opengl.GL11.glClear;
-import static org.terasology.rendering.opengl.OpenGLUtils.bindDisplay;
-import static org.terasology.rendering.opengl.OpenGLUtils.setViewportToSizeOf;
+import org.terasology.rendering.dag.stateChanges.BindFBO;
+import org.terasology.rendering.dag.stateChanges.Clear;
+import org.terasology.rendering.dag.stateChanges.EnableShader;
+import org.terasology.rendering.dag.stateChanges.SetViewportSizeOf;
 import static org.terasology.rendering.opengl.OpenGLUtils.renderFullscreenQuad;
 
 /**
  * TODO: Add diagram of this node
  */
 public class ToneMappingNode extends AbstractNode {
-    @In
-    private FrameBuffersManager frameBuffersManager;
-
-    @In
-    private WorldRenderer worldRenderer;
-
-    @In
-    private Config config;
-
-    private Material toneMapping;
-    private FBO sceneToneMapped;
-    private FBO sceneOpaque;
 
     @Override
     public void initialise() {
-        toneMapping = worldRenderer.getMaterial("engine:prog.hdr"); // TODO: rename shader to toneMapping)
+        addDesiredStateChange(new EnableShader("engine:prog.hdr"));
```

Comment:
Let's rename this to EnableMaterial. A material is the shader + all its configuration parameters.